### ISSUE TITLE
How to fix the working environment when running the nf-core/mag pipeline?

### ISSUE THREAD

**Issue by rbguevarra5** [nf-core/mag#430]
### Description of the bug

I am running the nf-core/mag pipeline in a cluster and I am using CONDA profile as my environment manager. Every time I run the pipeline in multiple jobs, it creates a new conda working environment for each of these jobs that I submit. I would like to fix the working directory so that the pipeline will not create conda environments from the beginning.

I have used the [-work-dir] argument in nextflow, and it fixes the working directory. However, the files stored in the working directories are mixed up with the files and folders from the previous run. I would like to have a clean working directory with the intermediate files and folders that is unique for each submission in separate folders.

Basically, I want to look for the .exitcodes of each run for every sample. But it is rather challenging to sort out when I have a bunch of projects running in parallel. In short, I want a unique 'work' folder for every run.

Is there a way to edit the configuration in nextflow to solve this issue?

Thank you. 

### Relevant files

![Screenshot 2023-05-02 at 14 33 55](https://user-images.githubusercontent.com/131269300/235682078-e00981fc-57c8-49df-8ce4-738a195ec6dd.png)

**Comment by jfy133** [comment]
Hi @rbguevarra5 there are a couple of questions there - I'll try to unpack them

1. For the first one, to re-use conda environments you need to set a conda cacheDir. You can do this either in a config file with [`conda.cacheDir`](https://www.nextflow.io/docs/latest/config.html#config-conda), or using teh bash environment variable [`NXF_CONDA_CACHEDIR`](https://www.nextflow.io/docs/latest/config.html#environment-variables).

2. For the separate work directories per run, that _should_ work with `-work-dir /path/to/unique/run/work`. If that's not working, then please try `-w`. However I don't understand 

    > files stored in the working directories are mixed up with the files and folders from the previous run
    
You might need to give some example commands - but you should run each `nextflow run` command in separate directories to make sure., and if `-w` is working correctly then you definitely should not be having them mixed up with the 'results' of the previous run..,.

**Comment by jfy133** [comment]
I will close this for now as it's a nextflow issue rather than nf-core/mag but feel free to conttinue leaving comments.
